src/dwarf_eh.h

```diff
@@ -237,9 +249,15 @@ static uint64_t read_value(char encoding, dw_eh_ptr_t *data)
 		READ(DW_EH_PE_sdata4, int32_t)
 		READ(DW_EH_PE_sdata8, int64_t)
 #ifdef __CHERI_PURE_CAPABILITY__
-		READ(DW_EH_PE_absptr, int64_t)
+        case DW_EH_PE_absptr:
+                {
+                        intptr_t t;
+                        memcpy(&t, *data, sizeof t);
+                        t = assert_pointer_in_bounds(t);
+                        return t;
+                }
 #else
-		READ(DW_EH_PE_absptr, intptr_t)
+		READ(DW_EH_PE_absptr, uintptr_t)
```

Comment:
Why has this changed?

Reply:
Just to be consistent with the rest of this file which was generally using uintptr_t instead of intptr_t.  I also changed the return type of this function from 'uint64_t' to 'uintptr_t' so having this be 'uintptr_t' seemed more consistent.